lib/SynchronousTransport.php

```
@@ -19,21 +19,27 @@ public function post(Environment $environment, $url, $payload)
         $environment->debug('posting to ' . $url, array('body' => $payload));
         $headers = array('Content-Type:application/json');
         if (curl_setopt($ch, CURLOPT_POSTFIELDS, $payload) === false) {
-            $environment->exception(new Exception('failed setting payload'));
             curl_close($ch);
-            return false;
+            $environment->exception(new Exception('failed setting payload'));
```

This should probably just throw the exception here instead of calling `$environment->exception`